### PR TITLE
Support for more gguf format and float zp for Q*_1

### PR DESCRIPTION
**Pull request by n1ck-guo**

### CODE REVIEW

File: auto_round/autoround.py [hunk 1]
```diff
@@ -162,6 +162,7 @@ def __init__(
             device_map: Union[str, dict] = None,
             super_bits: int = None,
             super_group_size: int = None,
+            float_zp: bool = False,
```
Comment (wenhuach21) [inline auto_round/autoround.py]: When the user only exports to GGUF using the q*1 format, switch to float zero points directly (add a new datatype), no need to add additional arguments to the API and wrapperlinear. If gptq/awq are also exported, using int zp and add a warning that this may affect the accuracy of gguf

File: auto_round/export/export_to_gguf/quant_cpu.py [hunk 2]
```diff
@@ -521,3 +536,37 @@ def q5_k_quant_block(blocks: np.array, scale=None, zp=None, wmin_m=None, d_scale
...
+
+    tmp_L = all_L.reshape(nb, 4, 64) & 0xF
+    output_ql = (tmp_L[:, ::2] | (tmp_L[:, 1::2] << 4)).reshape(nb, QK_K // 2)
+    output_qh = np.bitwise_or.reduce(
+        all_L.reshape(nb, 2, 4, 32) << np.array([0, 2, 4, 6]).reshape(1, 1, 4, 1), axis=2).reshape(nb, QK_K // 4)
+
+    output_d = output_d.reshape(-1, 1).astype(np.float16).view(np.uint8)
+
+    # [ql, qh, scale, d]
+    breakpoint()
```
Comment (wenhuach21) [inline auto_round/export/export_to_gguf/quant_cpu.py]: remember to delete later

File: auto_round/data_type/int.py [hunk 3]
```diff
@@ -72,13 +72,13 @@ def double_quant_tensor(tensor, bits, q_scale_thresh):
...
-def double_quant_tensor_sym(tensor, bits):
+def double_quant_tensor_sym(tensor, bits, q_scale_thresh):
     maxq = 2 ** (bits - 1)
     imax = abs(tensor).argmax(axis=-1, keepdims=True)
     wmax = torch.take_along_dim(tensor, imax, axis=-1)
-    iscale = torch.where(wmax !=0, -maxq / wmax, 0)
-    scale = torch.where(iscale != 0, 1 / iscale, 0)
-    qdq_tensor = torch.clamp(round_ste(tensor * iscale), min=-maxq, max=maxq-1) * scale
+    scale = wmax / -maxq
+    scale = torch.where(scale < 0, torch.clamp(scale, max=-q_scale_thresh), torch.clamp(scale, min=q_scale_thresh))
```
Comment (wenhuach21) [inline auto_round/data_type/int.py]: if the scale is always positive, no need to handle like this. Besides, it's mainly for fp16 datatype. If it's fp32 or bf16, I think handing zero scale is enough

File: auto_round/export/export_to_gguf/export.py [hunk 4]
```diff
@@ -23,17 +23,22 @@
...
-        "q4_k_s": gguf.LlamaFileType.MOSTLY_Q4_K_S,
-        "q2_k_s": gguf.LlamaFileType.MOSTLY_Q2_K_S,
+        "q5_0": gguf.LlamaFileType.MOSTLY_Q5_0,
+        "q5_1": gguf.LlamaFileType.MOSTLY_Q5_1,
         "q8_0": gguf.LlamaFileType.MOSTLY_Q8_0,
+        "q2_k_s": gguf.LlamaFileType.MOSTLY_Q2_K_S,
+        "q3_k_s": gguf.LlamaFileType.MOSTLY_Q3_K_S,
+        "q4_k_s": gguf.LlamaFileType.MOSTLY_Q4_K_S,
+        "q5_k_s": gguf.LlamaFileType.MOSTLY_Q5_K_S,
+        "q6_k": gguf.LlamaFileType.MOSTLY_Q6_K,
```
Comment (wenhuach21) [inline auto_round/export/export_to_gguf/export.py:40]: q6_k_s?

Comment (wenhuach21) [reply auto_round/export/export_to_gguf/export.py:40]: one is q6_k, the other one is q6_k_s， please have a double check

Comment (n1ck-guo) [reply auto_round/export/export_to_gguf/export.py:40]: There is no Q6_K_S, only Q6_K

File: auto_round/export/export_to_gguf/export.py [hunk 5]
```diff
@@ -23,17 +23,22 @@
...
+        "q6_k_s": gguf.LlamaFileType.MOSTLY_Q6_K,
```
Comment (wenhuach21) [inline auto_round/export/export_to_gguf/export.py:41]: then this one should be deleted?

File: test/test_gguf_format.py [hunk 6]
```diff
@@ -104,6 +104,51 @@ def test_q4_1(self):
...
+        quantized_model_path = "./saved"
+        autoround.quantize_and_save(output_dir=quantized_model_path, inplace=False, format="gguf:q*_1")
+        gguf_file = os.listdir("saved")[0]
+        model = AutoModelForCausalLM.from_pretrained(quantized_model_path, gguf_file=gguf_file, device_map="auto")
+        text = "There is a girl who likes adventure,"
+        inputs = self.tokenizer(text, return_tensors="pt").to(model.device)
+        print(self.tokenizer.decode(model.generate(**inputs, max_new_tokens=10)[0]))
+        shutil.rmtree("./saved", ignore_errors=True)
+    
+        model_name = "Qwen/Qwen2.5-1.5B-Instruct"
```
Comment (wenhuach21) [inline test/test_gguf_format.py:128]: better use 0.5b

Comment (n1ck-guo) [reply test/test_gguf_format.py:128]: The weight shape of 0.5b cannot be divided by 256, so q*k does not support 0.5b

File: test/test_gguf_format.py [hunk 7]
```diff
@@ -104,6 +104,51 @@ def test_q4_1(self):
...
+            self.tokenizer,
+            bits=3,
+            group_size=16,
+            sym=True,
+            iters=0,
+            data_type="int_sym_dq",
+            super_group_size=16,
+            super_bits=6
+        )
+        quantized_model_path = "./saved"
```
Comment (wenhuach21) [inline test/test_gguf_format.py:141]: add one more ut to cover asym/sym respectively

File: test/test_gguf_format.py [hunk 8]
```diff
@@ -104,6 +104,51 @@ def test_q4_1(self):
...
+            self.model,
+            self.tokenizer,
+            bits=bits,
+            group_size=group_size,
+            sym=sym,
+            iters=1,
+            data_type="int_asym_float_zp"
+        )
+        quantized_model_path = "./saved"
+        autoround.quantize_and_save(output_dir=quantized_model_path, inplace=False, format="gguf:q*_1")
```
Comment (wenhuach21) [inline test/test_gguf_format.py:118]: add q_40 ut if not present

**Comment by wenhuach21** [comment]
TestAutoRoundMLLM.test_quant_block_names 
the ut failure is related to new version of Transformers, you could fix in you new pr.

**Review by wenhuach21 — APPROVED** [review]
